### ISSUE TITLE
Extra Separator in Context Menu

### ISSUE THREAD

**Issue by deathernater** [atom/atom#5390]
![bug report](https://cloud.githubusercontent.com/assets/6615815/6034741/7cb2810a-abf4-11e4-8701-8d2c7b4b9570.png)

Is this extra bit at the end of the right-click context menu intended?

**Comment by benogle** [comment]
The menu algo really should check for separators before creating the menus.

**Comment by 50Wliu** [comment]
:+1: Really minor, but it'd be nice to have this fixed :).

**Comment by mnquintana** [comment]
:+1: I've reproduced this on both Windows 8.1 and Ubuntu 14.04 / 14.10.

**Comment by izuzak** [comment]
Does anyone see this problem on OSX? (I don't, but just want to make sure I'm not missing anything.)

![screen shot 2015-02-09 at 13 57 42](https://cloud.githubusercontent.com/assets/38924/6106719/3d58bfe6-b064-11e4-9ba6-db8e81cc92d8.png)

**Comment by mnquintana** [comment]
@izuzak Nope, it's just a Windows & Linux issue.

**Comment by deathernater** [comment]
Is there a way to manually fix this by changing some sort of style sheet?

**Comment by mnquintana** [comment]
I think this is because of the way separators in menus are defined currently - ie. manually with `{ type: 'separator' }`. When a menu gets added to the end of the context menu, if it happens to end with `{ type: 'separator' }`, then on Windows and Linux, a separator gets rendered anyway at the end of the context menu. 

IMO I feel like the best solution for this would be something like what was proposed in this Discuss thread: https://discuss.atom.io/t/is-there-a-way-to-specify-menu-item-order-when-inserting-menu-items/14937/11?u=mnquintana

```cson
{
  id: 'file'
  label: 'File'
  submenu: [
    { label: 'New Window', command: 'application:new-window' }
    { label: 'New File', command: 'application:new-file' }
    { label: 'Open...', command: 'application:open' }
    { label: 'Add Project Folder...', command: 'application:add-project-folder' }
    { label: 'Reopen Last Item', command: 'pane:reopen-closed-item' }
    { 'save-group': [
      { label: 'Save', command: 'core:save' }
      { label: 'Save As...', command: 'core:save-as' }
      { label: 'Save All', command: 'window:save-all' }
    ]}
    { 'close-group': [
      { label: 'Close Tab', command: 'core:close' }
      { label: 'Close Pane', command: 'pane:close' }
      { label: 'Close Window', command: 'window:close' }
    ]}
  ]
}
```

Instead of manually specifying separators, you would define groups, and Atom would then create separators as needed between those groups (and not at the end of the context menu).

**Comment by deathernater** [comment]
Any updates on whether this issue will be addressed soon? My OCD is getting the best of me haha.

**Comment by joshas** [comment]
:+1: Would be neat to have this fixed.

**Comment by damieng** [comment]
Fixed in Atom 1.11  (currently in beta)
